cpp/roaring.hh

```diff
@@ -45,6 +45,18 @@ class Roaring {
         }
         roaring.copy_on_write = r.roaring.copy_on_write;
     }
+    /**
+     * Move constructor
+     */
+    Roaring(Roaring &&r) {
+        bool is_ok = ra_init(&roaring.high_low_container);
+        if (!is_ok) {
+            throw std::runtime_error("failed memory alloc in constructor");
+        }
+        roaring.copy_on_write = false;
+        
+        std::swap(roaring, r.roaring);
```

Comment:
> Transfer of bare pointers means setting them NULL on the right side.

Is this true? So if I type ``uint32_t* y = std::move(x);``, then ``x``will be NULL?